### PR TITLE
WIP: Add option for file lookup to be within cwd (fixes #2736)

### LINKED ISSUE #2736: Alter config lookup?
this was brought up in the chat room, and I want to make sure we don't lose the discussion.there still seems to be a lot of confusion around configuration look up and how far it does or should go up to the directory structure. We had up until this point I looked all the way up to the home directory for configuration files. We would only use a configuration file in the home directory if there were no other configuration files found in subdirectories.

the current way this works was implemented here: https://github.com/eslint/eslint/issues/963

there was a discussion about whether or not we should keep looking up the directory structure or if we should stop at the current working directory. We eventually resolved to go all the way up to the home directory: https://github.com/eslint/eslint/issues/939#issuecomment-44234512

The question now is whether or not that choice was correct, and if it was correct, do we want a way to allow people to opt-out and specify where they want the lookup to stop.

The suggestion in the chatroom was to add "root": true to a config to say " don't look up any further from here. "

What do people think?

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: docs/developer-guide/nodejs-api.md [hunk 1]
```diff
@@ -79,6 +79,7 @@ The `CLIEngine` is a constructor, and you can create a new instance by passing i
 * `rulePaths` - An array of directories to load custom rules from (default: empty array). Corresponds to `--rulesdir`.
 * `rules` - An object of rules to use (default: null). Corresponds to `--rule`.
 * `useEslintrc` - Set to false to disable use of `.eslintrc` files (default: true). Corresponds to `--no-eslintrc`.
+* `root` - Set to true to enable file lookup to be within cwd (default: false). Corresponds to `--root`.
```
Comment (nzakas) [inline docs/developer-guide/nodejs-api.md]: Should be "set to true to stop configuration cascading from continuing to the parent directory"

**Comment by nzakas** [comment]
I think you misunderstood the issue. The change was to add a "root" option to .eslintrc files that would say "don't continue to the parent directory". There wouldn't be a couldn't line option.

**Comment by gyandeeps** [comment]
So `root` option should only be accepted through `eslintrc` file only and not through `cli` and `CLIEngine`? correct?
Just making sure I understand it correctly.

**Comment by nzakas** [comment]
Yup. You should be able to add:

```
root: true
```

In your .eslintrc file. That will tell us to stop searching any further up the file system for more .eslintrc files.

**Comment by gyandeeps** [comment]
@nzakas I dont think that would work out how everyone (i believe) wants. Can you please look at my comment on the issue. 
And also it will not work correctly as we merge configs files.

**Comment by nzakas** [comment]
OK, let's close this and continue discussion on the issue

**Pull request closed without merging**